### PR TITLE
Add play buttons to sound library tiles for touch

### LINKED ISSUE #4925: Cannot preview a sound on a touch device
### Expected Behavior

You should be able to preview sounds on a touch device. 

### Actual Behavior

We preview the sounds on `hover` so there is no way to preview sounds other than selecting them.

### PR DESCRIPTION
**Pull request by chrisgarrity**
### Resolves

_What Github issue does this resolve (please include link)?_

- Resolves #4925 

### Proposed Changes

Add prop to library items to show a play button. If a play button is present the ‘play on hover’ only applies over the icon rather than the full tile.

### Reason for Changes

We need a way to preview a sound on touch devices. There was also community feedback that it would be good to reduce the 'hover' area for sounds so they triggered a bit less.

Design Questions:
* Sounds start/stop on Mouse Enter/Leave the icon div. Should it stop on leaving the icon or on leaving the tile?
* How does the play button feel? Too big? Too small? Position?
* RTL: The location of the play button switches, but the direction of the arrow doesn't. This is consistent with the play button in the sound editor. Would it be better to flip in this context?

### Test Coverage

Current tests still run.

### Browser Coverage
Check the OS/browser combinations tested (At least 2)

Mac
 * [x] Chrome 
 * [ ] Firefox 
 * [x] Safari
 
Windows
 * [ ] Chrome 
 * [ ] Firefox 
 * [ ] Edge
 
Chromebook
 * [ ] Chrome
 
iPad
* [ ] Safari

Android Tablet
* [ ] Chrome

### CODE REVIEW

**Comment by rschamp** [comment]
Try it out: [sound-lib-touch](https://llk.github.io/scratch-gui/sound-lib-touch)

_I'm a bot, not actually @rschamp!_

**Comment by chrisgarrity** [comment]
Unfortunately the deploy step that makes the gh-pages branch is failing due to timing out - we probably have too many gh-pages branches. However, you can still try the branch out at https://chrisgarrity.github.io/scratch-gui/sound-lib-touch/

**Comment by chrisgarrity** [comment]
@carljbowman Revised the styles. New sound library items:
Screen Shot 2019-06-25 at 4 59 07 PM

**Comment by chrisgarrity** [comment]
After trying this with the community team there were several good suggestions for providing feedback about what is playing. Changing to needs work to address those concerns. /cc @carljbowman

**Comment by chrisgarrity** [comment]
@rschamp I'm pairing with Carl next week to make any styling tweaks we decide we want, but in the meantime this still needs a code review.

File: src/components/library/library.jsx [hunk 1]
```diff
@@ -69,22 +71,56 @@ class LibraryComponent extends React.Component {
         this.props.onRequestClose();
     }
     handleTagClick (tag) {
-        this.setState({
-            filterQuery: '',
-            selectedTag: tag.toLowerCase()
-        });
+        if (this.state.playingItem === null) {
+            this.setState({
+                ilterQuery: '',
```
[the remark this answers is not in the document]
Comment (chrisgarrity) [reply src/components/library/library.jsx]: Thanks for catching!

File: src/components/library-item/library-item.jsx [hunk 2]
```diff
@@ -115,19 +116,30 @@ class LibraryItemComponent extends React.PureComponent {
...
+                        onMouseLeave={this.props.showPlayButton ? this.props.onMouseLeave : null}
+                    >
                         <img
                             className={styles.libraryItemImage}
                             src={this.props.iconURL}
                         />
                     </Box>
                 </Box>
                 <span className={styles.libraryItemName}>{this.props.name}</span>
+                {this.props.showPlayButton ? (
```
[the remark this answers is not in the document]
Comment (rschamp) [reply src/components/library-item/library-item.jsx]: Nope this file doesn't use that convention

**Pull request merged**